simulator/modules/fetch/bpu/t/unit_test.cpp

```diff
@@ -15,27 +15,29 @@ TEST_CASE( "Initialization: WrongParameters")
     CHECK_THROWS_AS( BaseBP::create_bp( "saturating_two_bits", 100, 20), BPInvalidMode);
 }
 
-//TEST_CASE( "Static, all branches not taken")
-//{
-//    auto bp = BaseBP::create_bp( "always_not_taken", 128, 16);
-//
-//    Addr PC = 28;
-//    Addr target = 12;
-//    
-//    CHECK_FALSE( bp->is_taken(PC) );
-//    CHECK( bp->get_target(PC) == target);
-//}
-
-//TEST_CASE( "Static, all branches taken")
-//{
-//    auto bp = BaseBP::create_bp( "always_taken", 128, 16);
-//
-//    Addr PC = 28;
-//    Addr target = 12;
-//    
-//    CHECK( bp->is_taken(PC) );
-//    CHECK( bp->get_target(PC) == target);
-//}
+TEST_CASE( "Static, all branches not taken")
+{
+    auto bp = BaseBP::create_bp( "always_not_taken", 128, 16);
+
+    Addr PC = 28;
+    Addr target = 12;
+
+    bp->update( BPInterface( PC, false, target));
```

Comment:
Am I right putting ``` false ``` because it is "not taken" ?

Reply:
That's the input.

The idea of static predictions is that whereas the input is (T, T, T or T, NT, NT, NT, or NT, T, NT, T), the output should be the same.
I advise to test more patterns.

Reply:
For example write
```
 bp->update( BPInterface( PC, false, target));
 bp->update( BPInterface( PC, true, target));
 bp->update( BPInterface( PC, true, target));
```
Did I get it right?

Reply:
Yes. Please add a new test case to check them in a new pull request — I'm going to merge that one as soon as CI passes.